### PR TITLE
Content update - How to add an event handler using code (user story 1878475)

### PR DESCRIPTION
**Pull request by v-trisshores**
@adegeo, this PR is ready for your review.

[Internal preview link](https://review.docs.microsoft.com/en-us/dotnet/desktop/wpf/events/how-to-add-an-event-handler-using-code?view=netdesktop-6.0&branch=pr-en-us-1287)

### CODE REVIEW

**Comment by adegeo** [comment]
Thanks! I'll review this today.

File: dotnet-desktop-guide/net/wpf/events/how-to-add-an-event-handler-using-code.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+- The <xref:System.Windows.UIElement.AddHandler%2A> method.
+
+VB supports event handler assignment using:
+
+- The <xref:System.Windows.UIElement.AddHandler%2A> method, together with the [AddressOf](/dotnet/visual-basic/language-reference/operators/addressof-operator) operator to reference the event handler.
+- The [Handles](/dotnet/visual-basic/language-reference/statements/handles-clause) keyword in the event handler definition. For more information, see [Visual Basic and WPF event handling](/dotnet/desktop/wpf/advanced/visual-basic-and-wpf-event-handling?view=netframeworkdesktop-4.8&preserve-view=true).
+
+> [!NOTE]
+> In the following example, the C# code uses the `+=` operator, and the VB code uses the `AddHandler` method with `AddressOf` operator. Choose the language selector at the top of the page to select either C# or VB for the code sample.
+
```
Comment (adegeo) [inline dotnet-desktop-guide/net/wpf/events/how-to-add-an-event-handler-using-code.md:38]: I don't really think this note is needed. Either way, they'll only see what is appropriate for their chosen language. I'm not sure what benefit it does to call it out.

```suggestion
```

File: dotnet-desktop-guide/net/wpf/events/how-to-add-an-event-handler-using-code.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+## Example
+
+The following example has a <xref:System.Windows.Controls.Button> named `ButtonCreatedByXaml` defined in XAML. The button's XAML attribute `Click="ButtonCreatedByXaml_Click"` specifies the event handler for the <xref:System.Windows.Controls.Primitives.ButtonBase.Click> event. That attribute is in the form of `<event name>="<event handler name>"`.
+
+:::code language="xaml" source="./snippets/how-to-add-an-event-handler-using-code/csharp/MainWindow.xaml" id="ButtonCreatedByXaml":::
+
+:::code language="csharp" source="./snippets/how-to-add-an-event-handler-using-code/csharp/MainWindow.xaml.cs" id="ButtonEventHandlers":::
```
Comment (adegeo) [inline dotnet-desktop-guide/net/wpf/events/how-to-add-an-event-handler-using-code.md:45]: I'm not a fan of how consecutive code blocks look in the published article. I think it's nice to add some text between them:

 ```suggestion
:::code language="xaml" source="./snippets/how-to-add-an-event-handler-using-code/csharp/MainWindow.xaml" id="ButtonCreatedByXaml":::

The event handler code:

:::code language="csharp" source="./snippets/how-to-add-an-event-handler-using-code/csharp/MainWindow.xaml.cs" id="ButtonEventHandlers":::
```

File: dotnet-desktop-guide/net/wpf/events/snippets/how-to-add-an-event-handler-using-code/csharp/MainWindow.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            stackPanel.Children.Add(ButtonCreatedByCode);
+
+            // Assign an event handler to the new button using the '+=' operator.
+            ButtonCreatedByCode.Click += new RoutedEventHandler(ButtonCreatedByCode_Click);
+
+            // Assign an event handler to the new button using the AddHandler method.
+            // AddHandler(ButtonBase.ClickEvent, new RoutedEventHandler(ButtonCreatedByCode_Click);
+
+            // Assign an event handler to the StackPanel using the AddHandler method.
+            // stackPanel.AddHandler(ButtonBase.ClickEvent, new RoutedEventHandler(ButtonCreatedByCode_Click));
```
Comment (adegeo) [inline dotnet-desktop-guide/net/wpf/events/snippets/how-to-add-an-event-handler-using-code/csharp/MainWindow.xaml.cs:42]: I'm not seeing what the point of this commented code is for (just the `stackPanel` one) in both the VB and C#?

**Review by adegeo — CHANGES_REQUESTED** [review]
Looks great. I've left you some comments to consider.

Thanks!!

File: dotnet-desktop-guide/net/wpf/events/how-to-add-an-event-handler-using-code.md (hunk 2 above)
Comment (v-trisshores) [reply dotnet-desktop-guide/net/wpf/events/how-to-add-an-event-handler-using-code.md]: Fixed.

File: dotnet-desktop-guide/net/wpf/events/snippets/how-to-add-an-event-handler-using-code/csharp/MainWindow.xaml.cs (hunk 3 above)
Comment (v-trisshores) [reply dotnet-desktop-guide/net/wpf/events/snippets/how-to-add-an-event-handler-using-code/csharp/MainWindow.xaml.cs]: Fixed, by uncommenting and explaining how a routed event can repeatedly trigger a common event handler as it traverses the element tree.

**Comment by v-trisshores** [comment]
@adegeo thanks for your comments - this article is ready for a second review. Since the main feature of routed events is element traversal, I've modified the example to exclude a simple example of that.

File: dotnet-desktop-guide/net/wpf/events/how-to-add-an-event-handler-using-code.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+description: Learn how to add an event handler in code-behind for an element in Windows Presentation Foundation (WPF).
+ms.date: "02/01/2021"
+dev_langs:
+  - "csharp"
+  - "vb"
+helpviewer_keywords:
+  - "event handlers [WPF], adding"
+  - "XAML [WPF], adding event handlers"
+---
+<!-- The acrolinx score was 99 on 02/01/2021-->
```
Comment (adegeo) [inline dotnet-desktop-guide/net/wpf/events/how-to-add-an-event-handler-using-code.md:12]: I'm sorry but this should be 100.... just kidding, good job!

**Review by adegeo — APPROVED** [review]

File: dotnet-desktop-guide/net/wpf/events/how-to-add-an-event-handler-using-code.md (hunk 4 above)
Comment (v-trisshores) [reply dotnet-desktop-guide/net/wpf/events/how-to-add-an-event-handler-using-code.md:12]: @adegeo thanks for the review :)

**Pull request merged**